v2/dtos/requests/event.go

```diff
@@ -23,14 +23,11 @@ type AddEventRequest struct {
 	Event              dtos.Event `json:"event" validate:"required"`
 }
 
-// NewAddRequest creates, initializes and returns an AddEventRequest which has no readings
-// Sample usage:
-//    request := NewAddRequest("myProfile", "myDevice")
-//    request.Event.AddSimpleReading("myInt32Resource", v2.ValueTypeInt32, int32(1234))
-func NewAddRequest(profileName string, deviceName string) AddEventRequest {
+// NewAddEventRequest creates, initializes and returns an AddEventRequests
+func NewAddEventRequest(event dtos.Event) AddEventRequest {
 	return AddEventRequest{
 		BaseRequest: common.NewBaseRequest(),
-		Event:       dtos.NewEvent(profileName, deviceName),
+		Event:       event,
```

Comment:
please also modify dtos.NewEvent and use it to auto-fill some fields, especially apiVersion

Reply:
I believe `dtos.NewEvent` has taken care of the required fields already:
https://github.com/edgexfoundry/go-mod-core-contracts/blob/6e13956c373f8385b66ccdf7b0d57d3383bd1502/v2/dtos/event.go#L34-L42